File: javascript/bob/bob.js
export const hey = (message) => {
  let msg = message.trim();

  if (msg === null || msg === "") {
    return "Fine. Be that way!";
  }
  if (msg.endsWith("?")) {
    if (isUpperCase(msg)) {
      return "Calm down, I know what I'm doing!";
    }
    return "Sure.";
  }
  if (isUpperCase(msg)) {
    return "Whoa, chill out!";
  }
  return "Whatever.";
};

function isUpperCase(str) {
  return str != str.toLowerCase() && str == str.toUpperCase();
}

File: elixir/bob/lib/bob.ex
defmodule Bob do
  defp upcase?(str) do
    String.upcase(str) == str && String.downcase(str) != str
  end

  def hey(input) do
    trimmed = input |> String.trim()

    cond do
      trimmed == "" ->
        "Fine. Be that way!"

      String.ends_with?(trimmed, "?") && upcase?(trimmed) ->
        "Calm down, I know what I\'m doing!"

      String.ends_with?(trimmed, "?") ->
        "Sure."

      upcase?(trimmed) ->
        "Whoa, chill out!"

      true ->
        "Whatever."
    end
  end
end
